README.md

```diff
@@ -1,3 +1,5 @@
+HDF5 version 1.14.2 released on 2023-08-11
```

Comment:
Already below past the dashed-line separator

Reply:
One of the h5repack tests parses the current version of HDF5 from the third word in the README file and breaks if this is moved.